Engine/Source/ScenePanel.cpp

```diff
@@ -74,6 +78,74 @@ void ScenePanel::Draw(int windowFlags)
 			}
 			ImGui::EndDragDropTarget();
 		}
+
+		//Change the Guizmo operation using W,E & R keywords
+		if (ImGui::IsKeyPressed(ImGuiKey_W))
+			mCurrentGuizmoOperation = ImGuizmo::TRANSLATE;
+		if (ImGui::IsKeyPressed(ImGuiKey_E))
+			mCurrentGuizmoOperation = ImGuizmo::ROTATE;
+		if (ImGui::IsKeyPressed(ImGuiKey_R))
+			mCurrentGuizmoOperation = ImGuizmo::SCALE;
+
+		ImVec2 windowPos = ImGui::GetWindowPos();
+		ImGuizmo::SetDrawlist();
+		ImGuizmo::SetRect(windowPos.x, windowPos.y, ImGui::GetWindowWidth(), ImGui::GetWindowHeight());
+
+		const ModuleCamera* camera = App->GetCamera();
+		float4x4 cameraView = camera->GetViewMatrix().Transposed();
+		float4x4 cameraProjection = camera->GetProjectionMatrix().Transposed();
+
+		GameObject* selectedGameObject = ((HierarchyPanel*)App->GetEditor()->GetPanel(HIERARCHYPANEL))->GetFocusedObject();
+
+		//If there's a selected object in the hierarchy and it's not the root
+		if (selectedGameObject && (selectedGameObject != App->GetScene()->GetRoot())) {
+			const float4x4* transform = &selectedGameObject->GetWorldTransform();
+			float4x4 modelMatrix = selectedGameObject->GetWorldTransform().Transposed();
+
+			//Draws the Guizmo axis
+			ImGuizmo::Manipulate(cameraView.ptr(), cameraProjection.ptr(), mCurrentGuizmoOperation, mCurrentGuizmoMode, modelMatrix.ptr(), NULL, useSnap ? &snap[0] : nullptr);
+
+			if (ImGuizmo::IsUsing()) {
+				GameObject* parent = selectedGameObject->GetParent();
+				float4x4 inverseParentMatrix = float4x4::identity;
+				if (parent != nullptr) {
+					const float4x4* parentTransform = &parent->GetWorldTransform();
+					inverseParentMatrix = parent->GetWorldTransform().Inverted();
+				}
+				float4x4 localMatrix = inverseParentMatrix * modelMatrix.Transposed();
+
+				float3 translation;
+				Quat rotation;
+				float3 scale;
+				localMatrix.Decompose(translation, rotation, scale);
+
+				switch (mCurrentGuizmoOperation) {
+				case ImGuizmo::TRANSLATE:
+					selectedGameObject->SetPosition(translation);
+					break;
+				case ImGuizmo::ROTATE:
+					selectedGameObject->SetRotation(rotation);
+					break;
+				case ImGuizmo::SCALE:
+					selectedGameObject->SetScale(scale);
+					break;
+				}
+				selectedGameObject->RecalculateMatrices();
+			}
+		}
+
+		float viewManipulateRight = windowPos.x + size.x;
+		float viewManipulateTop = windowPos.y;
+		float viewManipulateSize = 100;
+
+		//TODO: Find the way to only apply the LookAt when pressing the ViewManipulateCube, if not, it causes issues with the free movement camera
```

Comment:
I'd propose to mantain this TODO in case the sample code is useful, else I'd propose just to mantain the TODO.